src/pybind/mgr/prometheus/module.py

```diff
@@ -517,28 +517,29 @@ def get_pg_status(self):
 
         for pool in pg_summary['by_pool']:
             total = 0
+            reported_states = {}
```

Comment:
how about
```py
num_by_state = dict((state, 0) for state in PG_STATES)
num_by_state['total'] = 0

for state_name, count in pg_summary['by_pool'][pool].items():
  for state in state_name.split('+'):
    num_by_state[state] += count
  num_by_state['total'] += count

for state, num in num_by_state.items():
  self.metrics[f'pg_{state}'].set(num, (pool,))
```

Reply:
Thanks @tchaikov ,
I updated it. I used old format to still support older python versions as nowhere in this file f-strings has been used
And overlapped via try/except as was used also in this code (prevents from misdeclaration of PG_STATES)

As I'm not a Ceph developer my assumptions above can be wrong

Reply:
thank you @jsuchenia you are right! we need to have this fix in nautilus also, which does not use python3 yet.